ArrayOperations/Arrays.cs

```diff
@@ -0,0 +1,324 @@
+    /// <exception cref="ArgumentException">The destination array has no elements.</exception>
+    public static int[] InsertFirst(int[] input, int number)
+    {
+      if (input == null)
+      {
+        throw new ArgumentNullException(nameof(input));
+      }
+
+      if (input.Length == 0)
+      {
+        throw new ArgumentException("Value cannot be an empty collection.", nameof(input));
+      }
+      
+      const int index = 0;
+      var output = InsertAt(input, number, index);
+      return output;
+    }
+
+    /// <summary>
+    /// Inserts a new element at the end of the array.
+    /// </summary>
+    /// <param name="input">Destination array.</param>
+    /// <param name="number">The number to be inserted.</param>
+    /// <returns>A new array with it's size increased by one and a new element located at the end of the array.</returns>
+    /// <exception cref="ArgumentNullException">The destination array is not initialized.</exception>
+    /// <exception cref="ArgumentException">The destination array has no elements.</exception>
+    public static int[] InsertLast(int[] input, int number)
+    {
+      if (input == null)
+      {
+        throw new ArgumentNullException(nameof(input));
+      }
+
+      if (input.Length == 0)
+      {
+        throw new ArgumentException("Value cannot be an empty collection.", nameof(input));
+      }
+
+      var output = InsertAt(input, number, input.Length);
+      return output;
+    }
+    
+    /// <summary>
+    /// Removes an element from the specified index of the array.
+    /// </summary>
+    /// <param name="input">Source array.</param>
+    /// <param name="index">The position at which we remove an element from the array.</param>
+    /// <returns>A new array with the size decreased by one and without the element found at specified index.</returns>
+    /// <exception cref="ArgumentNullException">The source array is not initialized.</exception>
+    /// <exception cref="ArgumentException">The source array has no elements.</exception>
+    /// <exception cref="ArgumentOutOfRangeException">Destination index is outside the bounds of the array.</exception>
+    public static int[] RemoveAt(int[] input, int index)
+    {
+      if (input == null)
+      {
+        throw new ArgumentNullException(nameof(input));
+      }
+
+      if (input.Length == 0)
+      {
+        throw new ArgumentException("Value cannot be an empty collection.", nameof(input));
+      }
+
+      if (index < 0 || index >= input.Length)
+      {
+        throw new ArgumentOutOfRangeException(nameof(index));
+      }
+
+      var output = new int[input.Length - 1];
+      for (var i = 0; i < output.Length; i++)
+      {
+        output[i] = i < index ? input[i] : input[i + 1];
+      }
+
+      return output;
+    }
+    
+    /// <summary>
+    /// Removes the first element from the array.
+    /// </summary>
+    /// <param name="input">Source array.</param>
```

Comment:
Minor nit: Now you're calling it source array which also doesn't make sense.  I get there is some difference "I am removing an element from...source" vs "I am adding an element to...destination".  But still, I think I'd just say "The array to remove an element from" or something like that.  I'm not sure source or destination are the right terms.

Reply:
Writing comments is cumbersome but I think it's good practice to describe what I was trying to achieve. Thank you.

Reply:
In general people don't comment every method they write.  Usually they separate their project (mentally or otherwise) into a "public" portion, methods that are meant to be called by other people that use your code, and a private portion, helper methods that help implement the public methods.  For every 1 public method (that you have to comment) you might have 10-50 private methods (that people usually don't comment, or they do so only briefly).